addons/portal/wizard/portal_wizard.py

```diff
@@ -173,11 +173,7 @@ def action_revoke_access(self):
 
         # remove the user from the portal group
         if user_sudo and user_sudo.has_group('base.group_portal'):
-            # if user belongs to portal only, deactivate it
-            if len(user_sudo.groups_id) <= 1:
-                user_sudo.write({'groups_id': [(3, group_portal.id), (4, group_public.id)], 'active': False})
-            else:
-                user_sudo.write({'groups_id': [(3, group_portal.id), (4, group_public.id)]})
+            user_sudo.write({'groups_id': [(3, group_portal.id), (4, group_public.id)], 'active': False})
```

Comment:
@tde-banana-odoo hey

Was contacted over PM for this. The idea would be that if the user has the portal group, then it would be fine to remove all of their other groups when revoking access as opposed to trying to pick and choose.

The problem being some groups are added to portal users automatically, which are not checked here. This results in users not actually being archived and thus having access to the portal anyway.

Previous iteration of this PR added a getter on users to register these groups we may be ok with revoking. They were wondering if this approach would be ok.

Pretty touchy so I thought it'd be better if you just had a look (or someone else who knows stuff, like xmo)

Reply:
@tde-banana-odoo blip

Reply:
Seems the behavior was always this one since https://github.com/odoo/odoo/commit/0b699fe7fbcc826159c04f7175ab4fafd60e60a6 (but indeed it can change).